src/plugins/vis_builder/public/application/utils/handle_vis_event.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get } from 'lodash';
+import { ExpressionRendererEvent } from '../../../../expressions/public';
+import { VIS_EVENT_TO_TRIGGER } from '../../../../visualizations/public';
+import { UiActionsStart } from '../../../../ui_actions/public';
+
+export const handleVisEvent = async (
+  event: ExpressionRendererEvent,
+  uiActions: UiActionsStart,
+  timeFieldName?: string
+) => {
+  const triggerId = get(VIS_EVENT_TO_TRIGGER, event.name, VIS_EVENT_TO_TRIGGER.filter);
+  let context;
+
+  if (triggerId === VIS_EVENT_TO_TRIGGER.applyFilter) {
+    context = {
+      timeFieldName,
+      ...event.data,
+    };
+  } else {
+    context = {
+      data: {
+        timeFieldName,
+        ...event.data,
+      },
+    };
+  }
```

Comment:
Is there an opportunity to refactor UIActions so we can use the same payload shape in both cases? This could also be simplified to something like

```suggestion
  const isApplyFilter = triggerId === VIS_EVENT_TO_TRIGGER.applyFilter;
  const dataContext  {
    timeFieldName,
    ...event.data,
  };
  const context = isApplyFilter ? dataContext : { data: dataContext } 
```

Reply:
The reason for this pattern is because the listeners of these actions expect this shape. Since i did not want to modify the shape across the app since that might break other expereinces, i just kept the shape the same here. I do like your simplification though :)

Reply:
Yeah, I was more thinking about whether it might be useful to refactor UIActions in the future to make integrations like this a bit cleaner.